model/src/main/kotlin/Vulnerability.kt

```diff
@@ -94,3 +104,25 @@ data class Vulnerability(
         }
     }
 }
+
+/**
+ * A custom deserializer to support the deserialization of [Vulnerability] instances using an older format, in which
+ * detailed information was embedded into the class rather than externalized in [VulnerabilityReference] objects.
+ */
+private class VulnerabilityDeserializer : StdDeserializer<Vulnerability>(Vulnerability::class.java) {
+    override fun deserialize(p: JsonParser, ctx: DeserializationContext): Vulnerability {
+        val vulnerabilityNode = p.codec.readTree<JsonNode>(p)
+        val id = vulnerabilityNode["id"].textValue()
+
+        return if (vulnerabilityNode["references"] != null) {
+            val references =
+                jsonMapper.convertValue(vulnerabilityNode["references"], jacksonTypeRef<List<VulnerabilityReference>>())
```

Comment:
Nit: The way we'd usually format this is

    val references = jsonMapper.convertValue(
        vulnerabilityNode["references"], jacksonTypeRef<List<VulnerabilityReference>>()
    )

or

    val references = jsonMapper.convertValue(
        vulnerabilityNode["references"],
        jacksonTypeRef<List<VulnerabilityReference>>()
    )

Reply:
Done. Note that I deliberately did not add a blank line after this block because IMHO this would not fit to the structure of the if-else construct.